dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -667,6 +680,10 @@ private boolean createWorkflowsAndVersionsFromDockstoreYml(List<? extends Workfl
                         Workflow workflow = createOrGetWorkflow(workflowType, repository, user, workflowName, wf, gitHubSourceCodeRepo);
                         WorkflowVersion version = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow, latestTagAsDefault, yamlAuthors);
 
+                        // Automatically register a DOI if it's enabled and the workflow is published and the version is a valid tag
+                        if (canAutomaticallyCreateDockstoreOwnedDoi(workflow, version)) {
+                            automaticallyRegisterDockstoreOwnedZenodoDOI(workflow, version, user, this);
```

Comment:
This new DOI creation code appears to run in the same database transaction that will create the version.  So, if the transaction is rolled back or fails, a DOI will be created that references a version that doesn't exist in the Dockstore database.   Recommend that the DOI creation code is executed after successful version creation.